extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/AbstractGuacamoleTunnelService.java

```diff
@@ -392,7 +395,29 @@ private GuacamoleSocket getUnconfiguredGuacamoleSocket(
         public ConnectionCleanupTask(ActiveConnectionRecord activeConnection) {
             this.activeConnection = activeConnection;
         }
-        
+
+
+
+        /***
+         * @Author zero
+         * @Description 将活动会话记录里的 会话开始时间变更为GUAC_DATE,GUAC_TIME
+         * @InitDate 10:53 2019/10/12
+         * @Param [activeConnection, tokens]
+         **/
+        public ConnectionCleanupTask(ActiveConnectionRecord activeConnection,Map<String ,String > tokens) {
+            this.activeConnection = activeConnection;
+
+            String yearMonthDay = tokens.get("GUAC_DATE");
+            String hourMinSecMs = tokens.get("GUAC_TIME");
+            SimpleDateFormat simpleDateFormat=new SimpleDateFormat("yyyyMMddHHmmss-SSS");
```

Comment:
I'm not sure about using the token to set the `ActiveConnectionRecord` date - in mind, this should actually work the other way around - the token should (somehow) be set using the date generated for the `ActiveConnectionRecord`, making the connection record the reference point and not the other way around.

@mike-jumper @ceharris @jmuehlner Any opinions on this??